### PR TITLE
Add compatibility with 2021.1

### LINKED ISSUE #230: Update NVDARemote to support cancellable speech
NVDA will be making cancellable speech commands available by default in the 2021.1 release of NVDA. This will also be a backwards compatibility breaking release, with an upgrade to Python 3.8. As such NVDA remote will need to be updated in order to be able to continue to work with the latest version of NVDA. A full list of compatibility breaking changes is maintained in the change log for NVDA.

For more information about cancellable speech, please see: https://github.com/nvaccess/nvda/issues/11736
Also related: https://github.com/NVDARemote/NVDARemote/issues/228

### PR DESCRIPTION
**Pull request by seanbudd**
A packaged copy of the patched addon can be downloaded here to test with: 
[remote-2.3-20210608.zip](https://github.com/NVDARemote/NVDARemote/files/6612656/remote-2.3-20210608.zip)

Note: I have only tested this with `e7568b709383e24eed263418eeb8a7a8a9a70822` as the base for this with master. This has since had an updated master merged in, and new features may be incompatible with 2021.1. 

Fixes have been introduced following the [Changes for Developers of 2021.1](https://www.nvaccess.org/files/nvda/releases/2021.1beta1/nvda_2021.1beta1_changes.html). Currently awaiting testing, which I will mark via the checkboxes, to see if the fixes work and if any more are needed. 

The following changes have been introduced to make NVDA remote compatible with NVDA 2021.1:

1. Update the manifest to 2021.1 
    A full smoke test of NVDA remote is required for this including the following:
    - [x] a 2021.1b instance (with patched add-on) controlling/controlled by an 2020.4 instance (with released add-on). 
    - [x] a 2021.1b instance (with patched add-on) controlling/controlled by a 2021.1b instance (with patched add-on). 
    - [x] a 2019.3 instance (with patched add-on) controlling/controlled by a 2020.4 instance (with released add-on). 
    Testing the following functionality:
      - connecting/disconnecting
      - controlling and listening to a remote device
      - configuring options
      - sending Ctrl+Alt+Del (I can't get this to work on my devices in general)
      - muting the remote device
      - pushing the clipboard
      - copying the link to remote

1. Commands are only in `speech.commands` as of https://github.com/nvaccess/nvda/issues/12126
The serializer checks if a command is in `speech.commands`, and tries to retrieve this command directly from `speech`. Instead the `getattr` should retrieve the command from `speech.commands`.

1. ignore cancellable speech commands.
This fixes #230, #228 and https://github.com/nvaccess/nvda/issues/11736.
   - [x] Specific testing strategy is needed to confirm these issues are fixed.

   Cancellable speech commands were introduced in 2020.2, so code is required to maintain backwards compatibility, or the minimum tested version could be bumped up to 2020.2. 

1. update beenCanceled to False using a workaround due to https://github.com/nvaccess/nvda/issues/12395
It would be ideal to remove this code entirely, as the NVDA devs believe the state of speech should only be interacted with via an API. If appropriate, it might be worth creating an API function in NVDA to set this to False, and then calling that instead.

### CODE REVIEW

**Comment by zstanecic** [comment]
Hi,
Can you update your branch, so that it complies with the latest master, where the latest security changes were included along with localization updates?
Best,
Zvonimir

**Comment by seanbudd** [comment]
Done this @zstanecic, I'll have a review of those code changes as well but they will need to be tested. I've updated the packaged addon for testing.

**Comment by zstanecic** [comment]
Ok.
Can you fix the following when it comes to this PR?
1. open the options dialog of the addon
2. press delete all fingerprints
3. press the no button
in the log, we see the following traceback:
ERROR - unhandled exception (09:03:41.257) - MainThread (1832):
Traceback (most recent call last):
  File "C:\Users\ninos\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient\dialogs.py", line 247, in on_delete_fingerprints
    evt.skip()
AttributeError: 'CommandEvent' object has no attribute 'skip'

**Comment by zstanecic** [comment]
When the new addon is installed on the NVDA 2020.4, i get the following error, tested on another user's machine:
ERROR - globalPluginHandler.listPlugins (09:12:53.004) - MainThread (12180):
Error importing global plugin remoteClient
Traceback (most recent call last):
  File "globalPluginHandler.pyc", line 23, in listPlugins
  File "importlib\__init__.pyc", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\matid\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient\__init__.py", line 20, in <module>
    from .session import MasterSession, SlaveSession
  File "C:\Users\matid\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient\session.py", line 19, in <module>
    (versionInfo.version_year == 2020 and versionInfo.major >= 2)
AttributeError: module 'versionInfo' has no attribute 'major'

**Comment by zstanecic** [comment]
Note: controlling between old version of the client and the new one works.
tested with the clients 2020.4 as the b (controlled) and 2021.1 as the a) (controlling)

**Comment by seanbudd** [comment]
Thanks for testing @zstanecic, these have been fixed now.

**Comment by zstanecic** [comment]
Ok. 

Notify me again, so that I can do the another round of testing.

**Comment by seanbudd** [comment]
@zstanecic, done

**Comment by zstanecic** [comment]
Thanks.

I will do the smoke test and give you the results at 1:00 gmt+2

File: addon/globalPlugins/remoteClient/session.py [hunk 1]
```diff
@@ -14,8 +13,21 @@
 from . import connection_info
 import hashlib
 
+
+if not (
```
Comment (lukaszgo1) [inline addon/globalPlugins/remoteClient/session.py:17]: Can't this condition be simplified to just `hasattr(speech.commands, "_CancellableSpeechCommand")`?

Comment (seanbudd) [reply addon/globalPlugins/remoteClient/session.py:17]: For now, yes. The intention here was to be more explicit behind the API change so if the minimum version is ever bumped up, these can be removed. Happy to change it either way.

**Comment by fisher729** [comment]
Hi.

When using both NVDA 2020.4 and NVDA 2021.1beta2 with this patched add-on, and I try to connect using the nvdaremote.com host, then I try to generate a key, I get the following error:
ERROR - stderr (20:41:21.913) - Thread-61 (512):
Exception in thread Thread-61:
Traceback (most recent call last):
  File "threading.pyc", line 926, in _bootstrap_inner
  File "threading.pyc", line 870, in run
  File "C:\NVDA\userConfig\addons\remote\globalPlugins\remoteClient\transport.py", line 54, in run
    self.server_sock.connect(self.address)
  File "ssl.pyc", line 1172, in connect
  File "ssl.pyc", line 1163, in _real_connect
  File "ssl.pyc", line 1139, in do_handshake
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1091)

In addition to that, when I make my own key, it is saying that the certificate is not secure, or something to that effect. However, this doesn't happen with the older add-on.

**Comment by tech10** [comment]
@fisher729 See #235 which was recently merged, I suspect generating a key wasn't checked with the latest security update to the add-on.

This is also why you are getting security prompts for your server. In reality, before this, no prompt was given and no keys were verified as being secure.

**Comment by fisher729** [comment]
Hi.

Ok, understood. This is interesting though, as it is happening with
the common, and add-on recommended, NVDA Remote server at
nvdaremote.com. I suspect users may not like this.

**Comment by seanbudd** [comment]
Hi @fisher729,

I've built a version of this add-on using master, without compatibility for 2021.1. Can you test this with 2020.4 and confirm the issue still occurs? If so, I think fixing it is out of the scope of this PR. 
[remote-2.3-master-20210607.zip](https://github.com/NVDARemote/NVDARemote/files/6605183/remote-2.3-master-20210607.zip)

**Comment by fisher729** [comment]
Hi.

Yes, confirmed that it does happen with the build posted and and NVDA
2020.4. Alright, I guess this is out of the scope of the pull request.

**Comment by seanbudd** [comment]
> Hi. Yes, confirmed that it does happen with the build posted and and NVDA 2020.4. Alright, I guess this is out of the scope of the pull request.

Yep, it is worth creating a new issue to track that problem though, otherwise it may end up in the next NVDARemote release.

**Pull request merged**